TriforkSwiftExtensions/TriforkSwiftExtensions/Date/DateExtension.swift

```diff
@@ -105,9 +106,73 @@ public extension Date {
         }
         return date
     }
-
+    
+    var startOfYear: Date {
+        alignTo(resolution: .year)
+    }
+    
+    var startOfMonth: Date {
+        alignTo(resolution: .month)
+    }
+    
+    var startOfDay: Date {
+        alignTo(resolution: .day)
+    }
+    
+    var startOfHour: Date {
+        alignTo(resolution: .hour)
+    }
+    
+    var startOfMinute: Date {
+        alignTo(resolution: .minute)
+    }
+    
+    var startOfSecond: Date {
+        alignTo(resolution: .second)
+    }
+    
     func date(byAdding component: Calendar.Component, value: Int) -> Date {
         return Calendar.current.date(byAdding: component, value: value, to: self) ?? self
     }
+    
+    func date(bySubtracting component: Calendar.Component, value: Int) -> Date {
```

Comment:
please take the current calendar as a optional parameter with default value to Calendar.current (in case someone is using this with a non default calendar). 
```swift
func date(
    bySubtracting component: Calendar.Component,
    value: Int,
    withCalendar: Calendar = Calendar.current
) -> Date {
    withCalendar.date(
        byAdding: component,
        value: -value,
        to: self
    ) ?? self
}